pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -515,7 +515,7 @@ func addEndpointSubset(subsets []v1.EndpointSubset, pod *v1.Pod, epa v1.Endpoint
 			Ports:     []v1.EndpointPort{epp},
 		})
 		readyEps++
-	} else {
+	} else if shouldPodBeInEndpoints(pod) {
```

Comment:
Change looks correct, but I think I'd still like a catch-all else that does nothing but print a message.  Just so we can debug.